### PR TITLE
chore: update to ESM-only version of remark

### PR DESCRIPTION
**Pull request by Trott**
Close: https://github.com/nodejs/remark-preset-lint-node/issues/207

BREAKING CHANGE: update to ESM-only version of remark

### CODE REVIEW

**Comment by Trott** [comment]
@nschonni If you've got something ready to go or nearly ready to go, I can close this, especially if you have opinions about some of the things in here. But if not, hey, I think this works.

File: package.json [hunk 1]
```diff
@@ -3,11 +3,12 @@
   "version": "2.4.1",
   "description": "remark preset to configure remark-lint with settings for nodejs/node",
   "main": "index.js",
```
Comment (nschonni) [inline package.json:5]: Not sure if this is required, but I think I had flipped this in my (non-working) branch
```suggestion
  "main": "./index.js",
```

File: index.js [hunk 2]
```diff
@@ -1,25 +1,49 @@
...
+import remarkLintNofileNameOuterDashes from "remark-lint-no-file-name-outer-dashes";
+import remarkLintNoHeadingIndent from "remark-lint-no-heading-indent";
+import remarkLintNoMultipleToplevelHeadings from "remark-lint-no-multiple-toplevel-headings";
+import remarkLintNoShellDollars from "remark-lint-no-shell-dollars";
+import remarkLintNoTableIndentation from "remark-lint-no-table-indentation";
+import remarkLintNoTabs from "remark-lint-no-tabs";
+import remarkLintNoTrailingSpaces from "remark-lint-no-trailing-spaces";
+import remarkLintNodejsLinks from "./remark-lint-nodejs-links.js";
+import remarkLintNodejsYamlComments from "./remark-lint-nodejs-yaml-comments.js";
+import remarkLintProhibitedStrings from "remark-lint-prohibited-strings";
```
Comment (nschonni) [inline index.js:29]: Does this need the new release, or is it OK as-is?

**Review by nschonni — APPROVED** [review]

**Comment by nschonni** [comment]
I'm guessing this should be a v3 release?

File: index.js (hunk 2 above)
Comment (Trott) [reply index.js:29]: Should be OK as-is.

**Comment by MaledongGit** [comment]
PS: There's something wrong with the Windows platform when converting to EMS..., see: https://github.com/nodejs/nodejs.org/pull/4052

Maybe the path?

**Comment by Trott** [comment]
> PS: There's something wrong with the Windows platform when converting to EMS..., see: [nodejs/nodejs.org#4052](https://github.com/nodejs/nodejs.org/pull/4052)
> 
> Maybe the path? On Linx everything goes right.

It's working in https://github.com/nodejs/nodejs.org/pull/4062 so whatever it was appears to be resolved by the updates and/or config change there.

**Pull request merged**